scripts/kfctl.sh

```diff
@@ -121,78 +158,66 @@ createNamespace() {
 }
 
 if [ "${COMMAND}" == "init" ]; then
-	DEPLOYMENT_NAME=${WHAT}
+  DEPLOYMENT_NAME=${WHAT}
 
-    while [[ $# -gt 0 ]]; do
+  while [[ $# -gt 0 ]]; do
     case $1 in
-        -h | --help)
-            usage
-            exit
-            ;;
-        --platform)
-            shift
-            PLATFORM=$1
-            ;;
-        --project)
-            shift
-            PROJECT=$1
-            ;;
-        --skipInitProject)
-            SKIP_INIT_PROJECT=true
-            ;;
-        --email)
-            shift
-            EMAIL=$1
-            ;;
-      esac
-      shift
-	done
-
-	mkdir -p ${DEPLOYMENT_NAME}
-	# Most commands expect to be executed from the app directory
-	cd ${DEPLOYMENT_NAME}
-	createEnv
-	source ${ENV_FILE}
-	# TODO(jlewi): Should we default to directory name?
-	# TODO(jlewi): This doesn't work if user doesn't provide name we will end up
-	# interpreting parameters as the name. To fix this we need to check name doesn't start with --
-	if [ -z "${DEPLOYMENT_NAME}" ]; then
-  		echo "name must be provided"
-  		echo "usage: kfctl init <name>"
-  		exit 1
-	fi
-    if [ -d ${DEPLOYMENT_NAME} ]; then
-		echo Directory ${DEPLOYMENT_NAME} already exists
-		exit 1
-	fi
-
-	if [ -z "${PLATFORM}" ]; then
-  		echo "--platform must be provided"
-  		echo "usage: kfctl init <PLATFORM>"
-  		exit 1
-	fi
-	source "${ENV_FILE}"
-
-	# TODO(jlewi): How can we skip GCP project setup? Add a command line argument
-	# to skip it?
-	if [ "${PLATFORM}" == "gcp" ]; then
-	  if [ ! ${SKIP_INIT_PROJECT} ]; then
-	  	gcpInitProject
-	  fi
-	fi
+      -h | --help)
+        usage
+        exit
+        ;;
+      --platform)
+        shift
+        PLATFORM=$1
+        mkdir -p ${DEPLOYMENT_NAME}
```

Comment:
Why did you move the mkdir and other commands here?

Reply:
Because this `DEPLOYMENT_NAME` is really only used by `gcp` specific `init` calls and Also I needed to be inside that directory to make `createEnv $*` call and then later parse args.
So `mkdir` needs to be called either here or within `createEnv` function. 🙂

Reply:
I"m still not following.

Should the instructions

mkdir -p ${DEPLOYMENT_NAME}
cd ${DEPLOYMENT}
createEnv

be executed for all platforms not just GCP?

The directory ${DEPLOYMENT_NAME} is used as the top level directory for all the Kubeflow related config (e.g. env.sh) so I don't think its GCP specific.

Reply:
I see you're passing the command line arguments back to createEnv and then reparsing them there to persist them.

Can we just set a dictionary of environment variables here and then pass that to createEnv? That seems a little cleaner then parsing command line arguments in two places. We can also avoid putting logic in the argument parsing case statements.

Reply:
You mean do the whole arg parsing business here, and then pass them to `createEnv` with the parsed values?

We can but do we want that in this PR or a subsequent PR.

I was thinking of doing one PR to just go over all the shell scripts and enhance path detection, format and style, enhance user messages specially during failures and that sort of thing.

Let me know what you think.

Reply:
Yes that's what I meant. Specifically; I'd like to do flag parsing in a single location. Otherwise we will likely have problems later on because we update one location but not the other.

I'm ok with doing this in a follow on PR as long as you think you can to it in the next week or so. I'll LGTM this with a hold. If you want to submit this first and fix command line parsing later just remove the hold.

Reply:
Thanks @jlewi 
I'll do the next PR sooner than later this week.
I'll file a ticket and go from there.